### PR TITLE
Field: transform renamed to map_coordinates

### PR DESCRIPTION
**Pull request by Ablinne**
map_coordinates now preserves the integral by applying the jacobian of
the transformation

### CODE REVIEW

File: postpic/datahandling.py [hunk 1]
```diff
@@ -706,6 +734,10 @@ def mean(self, axis=-1):
         return ret
 
     def _integrate_constant(self, axes=None):
+        if not self.islinear():
+            warnings.warn("Using method='constant' in integrate which is only suitable "
```
Comment (skuschel) [inline postpic/datahandling.py]: This warning should be an exception as this method will otherwise yield a wrong result.

**Comment by Ablinne** [comment]
The new method map_axis_grid is a solution to #94

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,8 +1,12 @@
 # Changelog of postpic
 
 ## current master
-
-
+2017-10-10
```
Comment (skuschel) [inline CHANGELOG.md]: can you at first add a section `incompatible changes` please? This is the most important part and also needs to be described well enough for users to understand what adjustments they have to make to their code.

File: test/test_datahandling.py [hunk 3]
```diff
@@ -260,12 +264,88 @@ def test_pad(self):
...
+        # this calculates numerical approx of derivative and tests
+        # helper.approx_1d_jacobian_det
+        b = self.f2d.map_axis_grid(1, lambda x: 20*x**2).integrate().matrix
+        print(a, b)
+        self.assertTrue(np.isclose(a, b, rtol=0.04))
+
     def test_integrate(self):
         print('start f1d.integrate')
-        a = self.f1d.integrate()
+        a = self.f1d.integrate(method='constant')
```
Comment (skuschel) [inline test/test_datahandling.py]: can you check all implemented integration methods, please?

File: test/test_datahandling.py [hunk 4]
```diff
@@ -83,6 +84,9 @@ def setUp(self):
         m = np.reshape(np.arange(60), (4, 5, 3))
         self.f3d = dh.Field(m)
 
+        x, y = np.meshgrid(np.linspace(0,1,100), np.linspace(0,1,100), indexing='ij', sparse=True)
+        self.f2d_fine = dh.Field(np.sin(x)*np.cos(y))
```
Comment (skuschel) [inline test/test_datahandling.py]: maybe its easier to sample the function between 0 and 2pi or 4pi and then you know, that the integration tests should yield zero.

**Pull request merged**